votor/src/votor.rs

```diff
@@ -140,6 +146,124 @@ pub struct Votor {
+            authorized_voter_keypairs,
+            blockstore,
+            bank_forks,
+            cluster_info,
+            leader_schedule_cache,
+            rpc_subscriptions,
+            snapshot_controller,
+            bls_sender,
+            commitment_sender,
+            drop_bank_sender,
+            bank_notification_sender,
+            leader_window_notifier,
+            event_sender,
+            event_receiver,
+            own_vote_sender,
+            consensus_message_receiver: bls_receiver,
+        } = config;
+
+        let start = Arc::new((Mutex::new(false), Condvar::new()));
+
+        let identity_keypair = cluster_info.keypair().clone();
+        let has_new_vote_been_rooted = !wait_for_vote_to_start_leader;
+
+        // Get the sharable root bank
+        let sharable_banks = bank_forks.read().unwrap().sharable_banks();
+
+        let shared_context = SharedContext {
+            blockstore: blockstore.clone(),
+            bank_forks: bank_forks.clone(),
+            cluster_info: cluster_info.clone(),
+            rpc_subscriptions,
+            leader_window_notifier,
+            vote_history_storage,
+        };
+
+        let consensus_metrics = Arc::new(PlRwLock::new(ConsensusMetrics::new(
+            sharable_banks.root().epoch(),
+        )));
+        let voting_context = VotingContext {
+            vote_history,
+            vote_account_pubkey: vote_account,
+            identity_keypair: identity_keypair.clone(),
+            authorized_voter_keypairs,
+            derived_bls_keypairs: HashMap::new(),
+            has_new_vote_been_rooted,
+            own_vote_sender,
+            bls_sender: bls_sender.clone(),
+            commitment_sender: commitment_sender.clone(),
+            wait_to_vote_slot,
+            sharable_banks: sharable_banks.clone(),
+            consensus_metrics: consensus_metrics.clone(),
+        };
+
+        let root_context = RootContext {
+            leader_schedule_cache: leader_schedule_cache.clone(),
+            snapshot_controller,
+            bank_notification_sender,
+            drop_bank_sender,
+        };
+
+        let timer_manager = Arc::new(PlRwLock::new(TimerManager::new(
+            event_sender.clone(),
+            exit.clone(),
+            consensus_metrics,
+        )));
+
+        let event_handler_context = EventHandlerContext {
+            exit: exit.clone(),
+            start: start.clone(),
+            event_receiver,
+            timer_manager: Arc::clone(&timer_manager),
+            shared_context,
+            voting_context,
+            root_context,
+        };
+
+        let consensus_pool_context = ConsensusPoolContext {
+            exit: exit.clone(),
+            start: start.clone(),
+            cluster_info: cluster_info.clone(),
+            my_vote_pubkey: vote_account,
+            blockstore,
+            sharable_banks,
+            leader_schedule_cache,
+            consensus_message_receiver: bls_receiver,
```

Comment:
```suggestion
            consensus_message_receiver,
```

If we accept the change above.